ultralist/file_store.go

```diff
@@ -7,41 +7,45 @@ import (
 	"os"
 )
 
+// the filename to use
+const FILENAME = ".todos.json"
+
 // FileStore is the main struct of this file.
 type FileStore struct {
-	FileLocation string
-	Loaded       bool
+	Loaded bool
 }
 
 // NewFileStore is creating a new file store.
 func NewFileStore() *FileStore {
-	return &FileStore{FileLocation: "", Loaded: false}
+	return &FileStore{Loaded: false}
 }
 
 // Initialize is initializing a new .todos.json file.
 func (f *FileStore) Initialize() {
-	if f.FileLocation == "" {
-		f.FileLocation = ".todos.json"
-	}
-
-	_, err := ioutil.ReadFile(f.FileLocation)
-	if err == nil {
+	if f.LocalTodosFileExists() {
 		fmt.Println("It looks like a .todos.json file already exists!  Doing nothing.")
 		os.Exit(0)
 	}
-	if err := ioutil.WriteFile(f.FileLocation, []byte("[]"), 0644); err != nil {
+	if err := ioutil.WriteFile(FILENAME, []byte("[]"), 0644); err != nil {
 		fmt.Println("Error writing json file", err)
 		os.Exit(1)
 	}
 }
 
-// Load is loading a .todos.json file.
-func (f *FileStore) Load() ([]*Todo, error) {
-	if f.FileLocation == "" {
-		f.FileLocation = f.GetLocation()
+// Returns if a local .todos.json file exists in the current dir.
+func (f *FileStore) LocalTodosFileExists() bool {
+	dir, _ := os.Getwd()
+	localrepo := fmt.Sprintf("%s/.todos.json", dir)
```

Comment:
would `filepath.Join(dir, FILENAME)`  be more robust?

Reply:
yep it would.